vividus-http-client/src/main/java/org/vividus/http/client/HttpClientFactory.java

```diff
@@ -68,62 +75,70 @@ public IHttpClient buildHttpClient(HttpClientConfig config) throws GeneralSecuri
+        Optional.ofNullable(config.getLastRequestInterceptor()).ifPresent(builder::addRequestInterceptorLast);
+        Optional.ofNullable(config.getLastResponseInterceptor()).ifPresent(builder::addResponseInterceptorLast);
 
         builder.setRedirectStrategy(config.getRedirectStrategy());
-        builder.setRetryHandler(config.getHttpRequestRetryHandler());
-        builder.setServiceUnavailableRetryStrategy(config.getServiceUnavailableRetryStrategy());
-        Builder requestConfigBuilder = RequestConfig.custom();
-        requestConfigBuilder.setConnectionRequestTimeout(config.getConnectionRequestTimeout());
-        requestConfigBuilder.setConnectTimeout(config.getConnectTimeout());
-        requestConfigBuilder.setCircularRedirectsAllowed(config.isCircularRedirectsAllowed());
-        requestConfigBuilder.setSocketTimeout(config.getSocketTimeout());
-        Optional.ofNullable(config.getCookieSpec()).ifPresent(requestConfigBuilder::setCookieSpec);
-        builder.setDefaultRequestConfig(requestConfigBuilder.build());
-        builder.setDefaultSocketConfig(SocketConfig.copy(SocketConfig.DEFAULT)
-                .setSoTimeout(config.getSocketTimeout())
+        builder.setRetryStrategy(config.getHttpRequestRetryStrategy());
+        builder.setDefaultRequestConfig(RequestConfig.custom()
+                .setConnectionRequestTimeout(Timeout.ofMilliseconds(config.getConnectionRequestTimeout()))
+                .setCircularRedirectsAllowed(config.isCircularRedirectsAllowed())
+                .setCookieSpec(config.getCookieSpec())
                 .build());
-        builder.setDnsResolver(config.getDnsResolver());
         builder.useSystemProperties();
 
         HttpClient httpClient = new HttpClient();
         httpClient.setCloseableHttpClient(builder.build());
         if (config.hasBaseUrl())
         {
-            httpClient.setHttpHost(HttpHost.create(config.getBaseUrl()));
+            httpClient.setHttpHost(HttpHost.create(URI.create(config.getBaseUrl())));
         }
         httpClient.setSkipResponseEntity(config.isSkipResponseEntity());
         httpClient.setHttpResponseHandlers(Optional.ofNullable(config.getHttpResponseHandlers()).orElseGet(List::of));
         return httpClient;
     }
 
+    private SSLConnectionSocketFactory buildSslSocketFactory(SslConfig sslConfig) throws GeneralSecurityException
+    {
+        SSLConnectionSocketFactoryBuilder builder = SSLConnectionSocketFactoryBuilder.create();
+        createSslContext(sslConfig.isSslCertificateCheckEnabled()).ifPresent(builder::setSslContext);
+        if (!sslConfig.isSslHostnameVerificationEnabled())
+        {
+            builder.setHostnameVerifier(NoopHostnameVerifier.INSTANCE);
+        }
+        return builder.build();
+    }
+
     private Optional<SSLContext> createSslContext(boolean sslCertificateCheckEnabled) throws GeneralSecurityException
     {
-        String protocol = SSLConnectionSocketFactory.SSL;
```

Comment:
hmm just wondering if it can cause incompatible behavior, we user SSL by default, but the new default in the new http client can be tls1.3

Reply:
actually it works in another way, see test results: https://github.com/vividus-framework/vividus/blob/master/vividus-tests/src/main/resources/story/integration/SslSteps.story#L9

Reply:
this is just an example with our test site, other people might stop supporting old protocols